From 00afae4af8c5b9d48023b9ceabeef79b92fcf646 Mon Sep 17 00:00:00 2001
From: Torge Matthies <openglfreak@googlemail.com>
Date: Mon, 7 Aug 2023 11:04:35 +0200
Subject: [PATCH 10/15] wintrust: Initialize sftSystemTime member of
 CRYPT_PROVIDER_DATA.

Some functions deep below these two functions would otherwise use this value without it being initialized properly
beforehand, resulting in them thinking that it's the 1st of January 1601.

Signed-off-by: Torge Matthies <openglfreak@googlemail.com>
---
 dlls/wintrust/wintrust_main.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/dlls/wintrust/wintrust_main.c b/dlls/wintrust/wintrust_main.c
index 11111111111..11111111111 100644
--- a/dlls/wintrust/wintrust_main.c
+++ b/dlls/wintrust/wintrust_main.c
@@ -263,6 +263,7 @@ static LONG WINTRUST_DefaultVerify(HWND hwnd, GUID *actionID,
     provData = WINTRUST_AllocateProviderData();
     if (!provData)
         return ERROR_OUTOFMEMORY;
+    GetSystemTimeAsFileTime(&provData->sftSystemTime);
 
     ret = WintrustLoadFunctionPointers(actionID, provData->psPfns);
     if (!ret)
@@ -452,6 +453,7 @@ static LONG WINTRUST_CertVerify(HWND hwnd, GUID *actionID,
     provData = WINTRUST_AllocateProviderData();
     if (!provData)
         return ERROR_OUTOFMEMORY;
+    GetSystemTimeAsFileTime(&provData->sftSystemTime);
 
     ret = WintrustLoadFunctionPointers(actionID, provData->psPfns);
     if (!ret)
-- 
0.0.0

